From: Michael Davidsaver <mdavidsaver@gmail.com>
Date: Tue, 27 Dec 2016 15:36:54 -0500
Subject: fix memory leak in GC feature node

---
 src/arvgcfeaturenode.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/arvgcfeaturenode.c b/src/arvgcfeaturenode.c
index d140468..ea15300 100644
--- a/src/arvgcfeaturenode.c
+++ b/src/arvgcfeaturenode.c
@@ -415,6 +415,10 @@ arv_gc_feature_node_init (ArvGcFeatureNode *gc_feature_node)
 static void
 arv_gc_feature_node_finalize (GObject *object)
 {
+	ArvGcFeatureNode *node = ARV_GC_FEATURE_NODE(object);
+
+	g_free (node->priv->name);
+
 	parent_class->finalize (object);
 }
 
