From: Emmanuel Pacaud <emmanuel@gnome.org>
Date: Wed, 1 Mar 2017 16:47:49 +0100
Subject: gvinterface: fix memory leak

---
 src/arvgvinterface.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/arvgvinterface.c b/src/arvgvinterface.c
index 0fbf4f6..ccdf573 100644
--- a/src/arvgvinterface.c
+++ b/src/arvgvinterface.c
@@ -295,6 +295,7 @@ arv_gv_interface_device_infos_unref (ArvGvInterfaceDeviceInfos *infos)
 		g_free (infos->user_name);
 		g_free (infos->manufacturer);
 		g_free (infos->serial_number);
+		g_free (infos->model);
 		g_free (infos->mac_string);
 		g_free (infos);
 	}
